### DISCUSSION TITLE
[API Request] Add `seed` parameter to random queries

### DISCUSSION THREAD

**Discussion by kgarner7** [opensubsonic/open-subsonic-api#92] [API requests]
### Type of change

API extension

### Proposal description

Currently, there are a few API endpoints that support sorting by random: `getAlbumList`, `getAlbumList2` and `getRandomSongs` (also `getSimilarSongs` + `getSimilarSogns2` if not using Last.fm). However, it is difficult to ensure deterministic paging across multiple queries as there isn't a way to identify that multiple queries should be seeded the same.

This extension would give clients the ability to specify a `seed` across multiple calls to enable deterministic `random` sorting.

### Backward compatibility impact

None; servers that don't implement the `seed` can just ignore it.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

Add a new string query parameter `seed` to the following endpoints that takes effect if the sort is `random`:
1. `getAlbumList`
2. `getAlbumList2`

Add a new string query parameter `seed` to `getRandomSongs`.

It _might_ also make sense to add the `seed` to `getSimilarSongs` and `getSimilarSongs2`, although for servers that use data by Last.fm this seed is not really helpful.

The specific implementation on how the seed is used by the server for random sorting is server-defined. If the `seed` is not defined, then the server may assume that each call has a different seed.

**Comment by Tolriq** [comment]
I 100% agree on the need, but I think it's time to build the proper new endpoints for getalbums/songs/artists & co that will support all new things and field for filtering and proper paging.

Adding a new field specific to paging for a temporary thing that I hope we address in the new endpoints does not make sense IMO.
